services/s3/src/main/java/software/amazon/awssdk/services/s3/multipart/MultipartConfiguration.java

```diff
@@ -0,0 +1,201 @@
+    private final Long minimumPartSizeInBytes;
+    private final Long apiCallBufferSizeInBytes;
+
+    private MultipartConfiguration(DefaultMultipartConfigBuilder builder) {
+        this.thresholdInBytes = builder.thresholdInBytes;
+        this.minimumPartSizeInBytes = builder.minimumPartSizeInBytes;
+        this.apiCallBufferSizeInBytes = builder.apiCallBufferSizeInBytes;
+    }
+
+    public static Builder builder() {
+        return new DefaultMultipartConfigBuilder();
+    }
+
+    @Override
+    public Builder toBuilder() {
+        return builder()
+            .apiCallBufferSizeInBytes(apiCallBufferSizeInBytes)
+            .minimumPartSizeInBytes(minimumPartSizeInBytes)
+            .thresholdInBytes(thresholdInBytes);
+    }
+
+    /**
+     * Indicates the value of the configured threshold, in bytes. Any request whose size is less than the configured value will
+     * not
+     * use multipart operation
+     * @return the value of the configured threshold.
+     */
+    public Long thresholdInBytes() {
+        return this.thresholdInBytes;
+    }
+
+    /**
+     * Indicated the size, in bytes, of each individual part of the part requests. The actual part size used might be bigger to
+     * conforms to
+     * the maximum
+     * number of parts allowed per multipart requests.
+     * @return the value of the configured part size.
+     */
+    public Long minimumPartSizeInBytes() {
+        return this.minimumPartSizeInBytes;
+    }
+
+    /**
+     * The maximum memory, in bytes, that the SDK will use to buffer requests content into memory.
+     * @return the value of the configured maximum memory usage.
+     */
+    public Long apiCallBufferSizeInBytes() {
+        return this.apiCallBufferSizeInBytes;
+    }
+
+    /**
+     * Builder for a {@link MultipartConfiguration}.
+     */
+    public interface Builder extends CopyableBuilder<Builder, MultipartConfiguration> {
+
+        /**
+         * Configures the minimum number of bytes of the body of the request required for requests to be converted to their
+         * multipart equivalent. Only taken into account when converting {@code putObject} and {@code copyObject} requests.
+         * Any request whose size is less than the  configured value will not use multipart operation,
+         * even if multipart is enabled via {@link S3AsyncClientBuilder#multipartEnabled(Boolean)}.
+         * <p></p>
+         *
+         * Default value: 8 Mib
+         *
+         * @param thresholdInBytes the value of the threshold to set.
+         * @return an instance of this builder.
+         */
+        Builder thresholdInBytes(Long thresholdInBytes);
+
+        /**
+         * Indicates the value of the configured threshold.
+         * @return the value of the threshold.
+         */
+        Long thresholdInBytes();
+
+        /**
+         * Configures the part size, in bytes, to be used in each individual part requests.
+         * <p></p>
```

Comment:
Should be single `<p>` as well. You can check the generated javadoc page locally after a successful build.